modules/swagger-codegen/src/main/resources/swift3/Models.mustache

```diff
@@ -202,6 +205,20 @@ class Decoders {
             }{{/isEnum}}
             {{^isEnum}}instance.{{name}} = Decoders.decodeOptional(clazz: {{{baseType}}}.self, source: sourceDictionary["{{baseName}}"] as AnyObject?){{/isEnum}}{{/allVars}}
             {{/unwrapRequired}}
+
+            {{#parent}}
+            for key in keys {
+                propsDictionary.removeValue(forKey: key)
+            }
+
+            for key in propsDictionary.keys {
+                if let decodedValue = Decoders.decodeOptional(clazz: String.self, source: propsDictionary[key] as AnyObject?) {
+                    instance[key as! String] = decodedValue
+                } else {
+                    instance[key as! String] = nil
```

Comment:
In my organization, we don't approve pull requests that use any of the force operators in Swift (`!` or `as!`). In general, why try to fight the type system? It's your friend. In this case, you're force casting the key to a string, which may get overlooked in future refactorings. It's not hard to make a change and I would be most appreciative :)